tests/unit/store/unload-test.js

```diff
@@ -84,9 +84,10 @@ test("unload a record", function(assert) {
 
       run(function() {
         store.unloadRecord(record);
+        assert.equal(get(record, 'hasDirtyAttributes'), false, "record is not dirty");
       });
 
-      assert.equal(get(record, 'hasDirtyAttributes'), false, "record is not dirty");
```

Comment:
@ryanpatrickcook so, we decided to change the implementation of the `hasDirtyAttributes` CP on the model, by aligning it to the implementation of the other flags. Can you change the dependent key [here](https://git.io/vrVBG) to be just `currentState` and add back the assertion. After that, this PR is good to go!